parrot/src/pfs_table.cc

```diff
@@ -654,7 +650,12 @@ pfs_file * pfs_table::open_object( const char *lname, int flags, mode_t mode, in
 			} else {
 				file = pname.service->open(&pname,flags,mode);
 			}
-			free(fd);
+		} else if(!pname.service->can_open_dirs() && (flags&O_DIRECTORY)) {
+			// This is a hack to maintain compatibility with the interpretation
+			// of O_DIRECTORY used in some of the services. If getdents fails
+			// on certain remote services or similar strangeness,
+			// this is a good place to start looking.
+			file = pname.service->getdir(&pname);
```

Comment:
Now that we have `can_open_dirs()`, we don't need to implement `getdir` for `local`, `cvmfs`, and `chirp`. I think it's safe to remove the those.